### ISSUE TITLE
Feat: Quick Draw

### ISSUE THREAD

**Issue by SunsetteSkye** [Vek17/WrathMods-TabletopTweaks#121]
Allows you to swap between weapon sets as a swift or move action.

**Comment by Amariithynar** [comment]
It's already a move action, though? Should be a free action as part of a move once you get to BAB+1, though, which isn't modelled in game (disable full round actions on weapon swap but don't consume move action, if possible) which this feat would make a free action in all circumstances, by RAW.

**Comment by Lord451125** [comment]
What you're talking about is just drawing a weapon.  That's free as part of a move, but in this drawing itself is a free action.  I believe what he's talking about is switching weapon groups which takes up your standard if I recall correctly.  Which means if you swap from your sword and board to a two hander, you can't attack.

**Comment by Amariithynar** [comment]
> What you're talking about is just drawing a weapon. That's free as part of a move, but in this drawing itself is a free action. I believe what he's talking about is switching weapon groups which takes up your standard if I recall correctly. Which means if you swap from your sword and board to a two hander, you can't attack.

No, I'm talking about switching weapons; it's a move action if you haven't used it yet. It then "upgrades" to a Standard if you don't have a move action available anymore. I then further stated that it SHOULD be a free action *as part of a move* (as in, so long as you move it shouldn't then consume your standard) by default, and that this feat in TT just outright makes it free, period. It's much more useful to use at range, but still restrictive.

**Comment by Skarm-Ghelmaad** [comment]
Would be a great addition, but I think that restricting drawing as "free as part of move" is a moot point that it is probably hard to implement in a CRPG, because there is no such a thing as a "free as part of move action" action in this game:

When you click is either one swift, one free, one move, one standard or one full-round action.

That official ruling was essentially an "artsy" idea developed by someone that absolutely lack the logical rigor that programming a machine to do something would require!

Essentially was a **pseudo-balancing cumbersome idea**!

In my opinion would be well enough to have a Quick Draw feat added which works as in Proper Flanking 2:

It is a move action unless you have Quick Draw...in which case becomes a free action!!

I'd really love this mod in this game too! :)

**Comment by Skarm-Ghelmaad** [comment]
> > What you're talking about is just drawing a weapon. That's free as part of a move, but in this drawing itself is a free action. I believe what he's talking about is switching weapon groups which takes up your standard if I recall correctly. Which means if you swap from your sword and board to a two hander, you can't attack.
> 
> No, I'm talking about switching weapons; it's a move action if you haven't used it yet. It then "upgrades" to a Standard if you don't have a move action available anymore. I then further stated that it SHOULD be a free action _as part of a move_ (as in, so long as you move it shouldn't then consume your standard) by default, and that this feat in TT just outright makes it free, period. It's much more useful to use at range, but still restrictive.

I think how it is implemented in TT is the most reasonable implementation for a computer game.

That canon ruling was just a moot point...just a bloody incoherent pseudo-balancing idea that must have been developed by some "artsy" guy completely devoid of **logical coherence**, which is something a machine would **require**:

Humans would leave flexibility for such a silly pseudo-coherent rules.

The machine would want you to tell him which action is...."free as part of a move action" would need to be a **specific kind of action** and since you **cannot** click to move while switching the weapon **simultaneously** (as the actions would be queued), it would be probably very complex to implement...

...all to ensure coherent with a pseudo-balancing rule that was **rubbish** to start with!

**Comment by Amariithynar** [comment]
>The machine would want you to tell him which action is...."free as part of a move action" would need to be a specific kind of action and since you cannot click to move while switching the weapon simultaneously (as the actions would be queued), it would be probably very complex to implement...

Not at all. If move action consumed this turn to move, switching weapons = free.

>...all to ensure coherent with a pseudo-balancing rule that was rubbish to start with!

Snipped just this line because it deals with everything you've said above otherwise. It isn't rubbish, nor is it a "pseudo-balancing rule". It represents how someone who has no martial training can't draw and use a weapon effectively at all in a short amount of time (ie, a single 6 second round) without just focusing on drawing the weapon and striking. A free draw *as part of a move* for BAB+1 and higher represents the martial training that the character has allowing them to split their focus and draw their weapon as part of moving into position to fight. Quick Draw as a feat represents someone taking that concept above and beyond what a fighter would need intrinsically, and learn how to draw weapons lightning-fast no matter the stance or situation.

**Comment by Mira132** [comment]
I think what you want is lightning swap and not quick draw since lightning swap is the feat that allows you to switch to different weapons using a free action instead of a move action. 

In WotR and KM, you are already considered as having quick draw since it would otherwise take a full round action to swap from one weapon set to another because sheathing a weapon is a move action and taking another out is also a move action. With quick draw, you can sheathe for a move action and draw as a free action resulting in the behaviour we currently have in-game.

**Comment by Skarm-Ghelmaad** [comment]
Well, but in a computer game would be unnecessarily cumbersome to do the "_full round action_" thing...personally I'd not consider a "plus" such a degree of realism...and the free action only to draw is a pretty stupid limitation that Pathfinder developers have introduced:

Quick Draw is an obvious reference to quick drawing a pistol or the iaijutsu "draw and cut" moves of Japanese swordsmanship and someone who can draw quickly can draw, cut and sheathe in a single motion...

So, in essence, it would be unrealistic (because the canon feat ruling is simply dumb and is not a good portrayal of the original "inspirations") and would be bothersome to play... :(

**Comment by Skarm-Ghelmaad** [comment]
> > The machine would want you to tell him which action is...."free as part of a move action" would need to be a specific kind of action and since you cannot click to move while switching the weapon simultaneously (as the actions would be queued), it would be probably very complex to implement...
> 
> Not at all. If move action consumed this turn to move, switching weapons = free.
> 
> > ...all to ensure coherent with a pseudo-balancing rule that was rubbish to start with!
> 
> Snipped just this line because it deals with everything you've said above otherwise. It isn't rubbish, nor is it a "pseudo-balancing rule". It represents how someone who has no martial training can't draw and use a weapon effectively at all in a short amount of time (ie, a single 6 second round) without just focusing on drawing the weapon and striking. A free draw _as part of a move_ for BAB+1 and higher represents the martial training that the character has allowing them to split their focus and draw their weapon as part of moving into position to fight. Quick Draw as a feat represents someone taking that concept above and beyond what a fighter would need intrinsically, and learn how to draw weapons lightning-fast no matter the stance or situation.

Nah...it is just rubbish...was just a very feeble attempt to "give something but not so much" to the characters with martial training. As for Quick Draw, how Paizo developed it is as much stupid:

The obvious reference are the **[chanbara](https://en.wikipedia.org/wiki/Samurai_cinema)** and the **iaijutsu** techniques of Japanese swordsmanship, but those involve **drawing**, **cutting down the enemy** (ideally) and **sheathing** in a **single fluid motion**.

The problem is simple:

Paizo had too many fanboys "licking" them no matter how much sloppy were the rules they devised or the products they dished out, so they grew complacent...starting from a few rules that were incoherent and/or absurd they ended up nerfing the game to introduce feats that would **un-nerf** the game just to **cash in easy money** without putting a shred of brain in what they were offering to their audience!

P.S.: However you have a point on one thing:
There is already a game feature that works in a similar way:

Dimensional Slide is activated as a swift action but works as part of a move action.

However was correct to point out that what we are doing in-game is swapping weapons not just drawing and sheathing them.

**Comment by SunsetteSkye** [comment]
... Quick Draw was part of D&D 3. Before things got really out of hand with "rules clarifications" in D&D 3 and later PF , the feat was just meant to make it easy to switch weapons. Iaijitsu came later.

The feat is older than swift actions, that's why the action without the feat is expressed as a free action made as part of a move action. There was not better terminology and no one cared to update it properly. I don't care which additional absolutely unnecessary feats Pathfinder added later in order to make this, already situational and weak feat EVEN WEAKER.

**Comment by Skarm-Ghelmaad** [comment]
> ... Quick Draw was part of D&D 3. Before things got really out of hand with "rules clarifications" in D&D 3 and later PF , the feat was just meant to make it easy to switch weapons. Iaijitsu came later.
> 
> The feat is older than swift actions, that's why the action without the feat is expressed as a free action made as part of a move action. There was not better terminology and no one cared to update it properly. I don't care which additional absolutely unnecessary feats Pathfinder added later in order to make this, already situational and weak feat EVEN WEAKER.

Sorry, but **Iaijutsu** is not a feat, it is a type of martial arts technique and they existed **a lot before** Gary Gygax was in the womb of his mother...so it predates even **Chainmail**, **D&D 1st edition** and pretty much **everything** in the roleplaying game world!! 🤔

The point is simple:

Was something obviously **inspired** by that kind of thing, since in the West the only situation in which there was a focus on "drawing first" was about "drawing a gun" and was in the Far West...which wasn't **much related** to the typical **medieval fantasy** setting (at least before Pathfinder introduced the gunslinger!)...
